fd/diff.go

```diff
@@ -19,25 +18,30 @@ type Point struct {
 	Coeff float64
 }
 
-// Method represents a finite difference formula that approximates
-// the derivative of order k of a function f at x as
-//  d^k f(x) ≅ (1 / h^k) * \sum_i Coeff_i * f(x + h * Loc_i),
-// where h is a small positive step.
-type Method struct {
+// Formula represents a finite difference formula on a regularly spaced grid
+// that approximates the derivative of order k of a function f at x as
+//  d^k f(x) ≅ (1 / Step^k) * \sum_i Coeff_i * f(x + Step * Loc_i).
+type Formula struct {
 	// Stencil is the set of sampling Points which are used to estimate the
 	// derivative. The locations will be scaled by Step and are relative to x.
 	Stencil []Point
 	Order   int     // The order of the approximated derivative.
-	Step    float64 // Default step size for the method.
+	Step    float64 // Default step size for the formula.
```

Comment:
> Citation? The conclusion seems to be correct, I've just never heard a mathematical argument made that says that before and I'd be interested in reading it.

For example Nocedal&Wright, Chapter 8 (Calculating Derivatives). The derivation is pretty simple. For the first order derivatives, it gives optimal step ~eps^1/2 for the one-sided approximations and ~eps^1/3 for the centered. For the standard second derivative approximation, it gives an optimal step ~eps^1/4. Of course, assuming that the function is sufficiently smooth, etc.

> Lets put off the default behavior to the next PR.

Agreed.

> I do think Central is a good default ...

OK.